### ISSUE TITLE
Regression: Navigation setting issues in 3.4beta

### ISSUE THREAD

**Issue by sonora** [osmandapp/Osmand#7198]
Looks to me like we have an issue with the navigation settings in 3.4beta: Often or always not the "Fastest route" is calculated, but what looks like the "Fuel-efficient route" (or maybe shortest route), no matter what the settings say? (Driving mode, Base profile, nightly of July 1 and July 8.)

**Comment by vshcherb** [comment]
In debug it looks like it works. It is probably UI glitch of changing settings

**Comment by vshcherb** [comment]
https://github.com/osmandapp/OsmAnd-core/commit/97bc6989d7053a331def4de4abb74ad942273e29

**Comment by vshcherb** [comment]
Probably this is going to help, it fixes issues short distances

**Comment by sonora** [comment]
From what I see, it did not fix this.

For arbitrary routes of about 100 km I find the following:
(1) While the old "fastest routes" used to take the slightly longer highway (e.g. 120km, 1.05h), OsmAnd now calculates via primary roads, e.g. 110km, 1:41h
(2) UI:  The toast which appears when the route finishes calculating even announces **2**:41h, strangely
(3)- Also UI: Yes, there are some Settings inconsistencies or screen update issues, like e.g. the following:  On the route definition screen, hit the settings wheel, then hit Navigation settings: A change of "Fastest route" is not reflected when coming back to the route definition's settings screen, only after returning all the way to the rote definition screen and again tapping the settings wheel. But his issue seems not the one I am reporting here - which is more fundamental. I find noway to truly calculate the fastest route with new nightly (v3.4) builds.

**Comment by vshcherb** [comment]
(3) yes we've noticed requires a separate issue

Yes, I was fixing 2 major issues related to navigation they weren't in 3.3 but in between. I don't remember any big coefficients adjustments except then that we've debugged all parameters they look to be the same. 

2:41 is quite a big penalty comparing to 1:05 and 1:41. Please share the route it is important I think but I need to test the route itself. I don't think we've introduced a bug to the routing engine though I believe there is a huge penalty somewhere pushing it for 1 hour extra.

**Comment by sonora** [comment]
I found a route where it is pretty obvious: On the Germany_Bavaria map (I have the June 1, 2019 edition), just go from the town of Erlangen to the town of 'Weiden in der Oberpfalz', no further address specification each. The motorways between both towns form sort of  a U-shaped route and that was usually suggested, taking about 1 hour, maybe 1:10, can't remember. Osmand now claculates the route on the somewhat more direct primary road, announcing 1:46h. (Again the brief toast when the route calculation finishes announces even 2:33, but that is bogus).

**Comment by sonora** [comment]
If there were no significant changes to the routing engine, I guess a candidate for trouble could be the new min/max speed settings introduced. I have not changed them, they are at default.

**Comment by vshcherb** [comment]
I calculated both routes, so via motorway is actually longer "10063.79" vs "9126" seconds. I think there is no issue in routing engine but there is an issue in configuration looks like

**Comment by vshcherb** [comment]
Issue has found and fixed - https://github.com/osmandapp/OsmAnd-core/commit/141b6cf47a8f35d91c28bdb078276b9537706454. Indeed it was limiting high speed to default speed which is weird.

**Comment by sonora** [comment]
Confirmed working. Good that we caught it. ;)

 - There is still a mismatch between the time the toast announces,and the actual time the route details screen announces.

 - Also, the time calculated now for the same route is not the time the older OsmAnd versions estimated. Will monitor if they are better or worse.

 - Route  calculation seems to take longer than with prior versions, but I cannot quantify for now.

 - See also #7230. Many question marks with several users. ;)
